src/Microsoft.AspNetCore.OData/Batch/ODataBatchStream.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.IO;
+
+namespace Microsoft.AspNet.OData.Batch
+{
+    // Normally, the stream is closed when the ODataMessageWriter is disposed.
+    // For responses within a batch, we need to read from the stream after the response is
+    // written in order to write it to the batch response stream. So we need to ignore the Close()
+    // and provide an alternate method to release the stream after writing its content to the batch response.
+    internal class BatchStream : MemoryStream
+    {
+        /// <summary>
+        /// Release the batch stream and underlying resources
+        /// </summary>
+        internal void Release()
```

Comment:
>Release() [](start = 22, length = 9)

Detach ?

Reply:
Well, it's not actually detaching anything; the memory stream is fully contained by the ODataBatchStream; it's just releasing the resources associated with (underlying) stream.

Maybe Free()?  Discard? Internal_Dispose?

---
In reply to: [377796792](https://github.com/OData/WebApi/pull/2048#discussion_r377796792) [](ancestors = 377796792)

Reply:
Went with InternalDispose(), since that's really what's happening.

---
In reply to: [377919530](https://github.com/OData/WebApi/pull/2048#discussion_r377919530) [](ancestors = 377919530,377796792)